pages/docs/reference/whatsnew14.md

```diff
@@ -574,3 +574,306 @@ Complete the initial configuration, and when you add a new dependency to `cocoap
+    //sampleEnd
+        println("The most expensive item in the order costs $highestPrice")
+    }
+    ```
+    </div>
+
+    There are also `minOfWith()` and `maxOfWith()`, which take a `Comparator` as an argument, and `*OrNull()` versions
+of all four functions that return `null` on empty collections.
+
+    In turn, `min()` and `max()` functions have been renamed to `minOrNull()` and `maxOrNull()` to comply with the naming
+convention used across the Kotlin collections API: `*OrNull` suffix in the function name means that it returns `null`
+if the receiver collection is empty.
+
+* New overloads for `flatMap` and `flatMapTo` let you use transformations with return types that don’t match the receiver type, namely:
+    * transformations to `Sequence` on `Iterable`, `Array`, and `Map`
+    * transformations to `Iterable` on `Sequence`
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        val list = listOf("kot", "lin")
+        val lettersList = list.flatMap { it.asSequence() }
+        val lettersSeq = list.asSequence().flatMap { it.toList() }    
+    //sampleEnd
+        println(lettersList)
+        println(lettersSeq.toList())
+    }
+    ```
+    </div>
+
+* `flatMapIndexed()` counterpart for `flatMap()`.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+        val list = listOf("hello", "kot", "lin", "world")
+        val kotlin = list.flatMapIndexed { index, item ->
+            if (index in 1..2) item.toList() else emptyList() 
+        }
+    //sampleEnd
+        println(kotlin)
+    }
+    ```
+    </div>
+
+#### Arrays
+
+To provide a consistent experience when working with different container types, we’ve also added new functions for **arrays**:
+
+* `shuffle()` - puts the array elements in a random order.
+* `onEach()` - performs the given action on each array element and returns the array itself.
+* `reverse()` for array subranges reverses the order of the elements in the subrange.
+* `sortDescending()` for array subranges sorts the elements in the subrange in descending order.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+
+```kotlin
+fun main() {
+//sampleStart
+    var language = ""
+    val letters = arrayOf("k", "o", "t", "l", "i", "n")
+    val fileExt = letters.onEach { language += it }
+       .filterNot { it in "aeuio" }.take(2)
+       .joinToString(prefix = ".", separator = "")
+    println(language) // "kotlin"
+    println(fileExt) // ".kt"
+
+    letters.shuffle()
+    letters.reverse(0, 3)
+    letters.sortDescending(2, 5)
+    println(letters.contentToString()) // [k, o, t, l, i, n]
+//sampleEnd
+}
+```
+</div>
+
+### Converting from KType to Java Type
+
+A new extension property `KType.javaType` helps obtain a `java.lang.reflect`.Type from a Kotlin type without
```

Comment:
It's new in stdlib and experimental, because it doesn't cover all cases. The same named extension is available in `kotlin-reflect` and AFAIK works better, but requires `kotlin-reflect` dependency as a price.